core/domain/collection_services.py

```diff
@@ -482,6 +484,11 @@ def _save_collection(committer_id, collection, commit_message, change_list):
     memcache_services.delete(_get_collection_memcache_key(collection.id))
     index_collections_given_ids([collection.id])
 
+    is_public = rights_manager.is_collection_public(collection.id)
```

Comment:
Also I think it might be better to put this in update_collection() instead, which has other rights-related stuff